internal/cephfs/clone_test.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cephfs
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCloneStateToError(t *testing.T) {
+	errorState := make(map[cephFSCloneState]error)
+	errorState["cephFSCloneComplete"] = nil
+	errorState["cephFSCloneError"] = ErrInvalidClone
+	errorState["cephFSCloneInProgress"] = ErrCloneInProgress
+	errorState["cephFSClonePending"] = ErrClonePending
+	errorState["cephFSCloneFailed"] = ErrCloneFailed
+
+	assert.Equal(t, cephFSCloneComplete.toError(), errorState["cephFSCloneComplete"])
+	assert.Equal(t, cephFSCloneError.toError(), errorState["cephFSCloneError"])
+	assert.Equal(t, cephFSCloneInprogress.toError(), errorState["cephFSCloneInProgress"])
+	assert.Equal(t, cephFSClonePending.toError(), errorState["cephFSClonePending"])
+	assert.Equal(t, cephFSCloneFailed.toError(), errorState["cephFSCloneFailed"])
```

Comment:
That looks reasonable, but I would recommend doing it like:
```go
errorStates := make(map[cephFSCloneState]error)
errorStates[cephFSCloneComplete] = nil
errorStates[cephFSCloneError] = ErrInvalidClone
...
errorStates["someInvalidValue"] = noIdea // whatever expected error
...

for state, err := range errorStates {
    assert.Equal(t, state.toError(), err)
}
```

Callers are not expected to manually use the string values of `cephFSCloneState`, so that does not need to be tested with this function. (There could be an other function that does it.)

Reply:
okay.